pkg/cmd/cli/cmd/tag.go

```diff
@@ -317,94 +317,152 @@ func (o TagOptions) RunTag() error {
 
 		err := kclient.RetryOnConflict(kclient.DefaultRetry, func() error {
 			isc := o.osClient.ImageStreams(o.destNamespace[i])
-			target, err := isc.Get(destName)
-			if err != nil {
-				if !kerrors.IsNotFound(err) {
+
+			istag := &imageapi.ImageStreamTag{
```

Comment:
`istag` is not not used in case of deletion. Move it where you actually create the tag.